src/index.js

```diff
@@ -139,7 +141,11 @@ class TrustWeb3Provider {
   }
 
   eth_signTypedData(payload) {
-    this.postMessage("signTypedMessage", payload.id, {data: payload.params[1]});
+    this.postMessage("signTypedMessage", payload.id, {data: payload.params[0]});
+  }
+
+  eth_signTypedDataV3(payload) {
+    this.postMessage("signTypedMessageV3", payload.id, {data: payload.params[1]});
```

Comment:
I I suggest using `signTypedMessage` and `signTypedMessageV1` if you really want to support legacy 712 (not a standard spec)

Reply:
Got it